sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -92,6 +93,31 @@ object IntervalUtils {
   }
 
   private val yearMonthPattern = "^([+|-])?(\\d+)-(\\d+)$".r
+  private val yearMonthStringPattern =
+    "^(INTERVAL\\s+)([+|-])*?(')([+|-])?(\\d+)-(\\d+)(')(\\s+YEAR TO MONTH)$".r
```

Comment:
I still don't understand `([+|-])*?`. Do we support `INTERVAL ++'1-1' YEAR TO MONTH`?

Reply:
`YEAR TO MONTH` should use `\s+`

Reply:
> I still don't understand `([+|-])*?`. Do we support `INTERVAL ++'1-1' YEAR TO MONTH`?

Updated

Reply:
> `YEAR TO MONTH` should use `\s+`

Done